### DISCUSSION TITLE
Error fetching user group data - Post update to V3.0

### DISCUSSION THREAD

**Discussion by FirePower99** [Budibase/budibase#14975] [Help] [answered]
After upgrading to v3.0 on my self hosted instance of budibase, I have lost access to the main user management function.

No users are listed in the Users section: "Add users and control who gets access to your published apps" 
There is a red error message displayed on loading that page:  "Error fetching user group data"

I did follow the upgrade instructions prior to upgrading to v3.0  to bring couch_db - up to using image budibase/couchdb:v3.3.3-sqs-v2.1.1

And it was working fine before the upgrade to v3.0.

I have managed to subsequently upgrade now to v3.10 - but this error still exists. - Any help would bee greatly appreciated.

**Comment by melohagan** [comment]
Hey @FirePower99 

Thank you for reporting. Would you be able to provide some additional information please, as we are struggling to reproduce the issue.

What type of license do you have?
Can you show any screenshots / recordings?
How many groups do you have, and how many users in each group?
What roles do the groups have within your apps?

**Reply by FirePower99** [reply]
Hey @melohagan 

Thank you for getting back. Sure, let me see what I can provide:

I'm currently on a Premium Plan
budibaseVersion: 3.1.0
hosting: docker-compose
nodeVersion: v20.18.0
platform: linux
Couch_db: v. 3.3.3

I didn't have any groups setup. So I'm surprised this is throwing an error. I thought that was only an enterprise feature.

I do see the following error in the console when loading the user page:  HTTP 404 on GET : https://dcd-dataforge.com/api/global/tenant/default

I currently only have 3 users., 2 Basic Users, and 1 Admin.

This issue seems very similar to the one reported [here](https://github.com/Budibase/budibase/issues/14890)

<img width="955" alt="Screenshot 2024-11-06 120618" src="https://github.com/user-attachments/assets/d6f50e63-ac55-4346-8edc-96ea678ca292">

**Reply by melohagan** [reply]
Thanks for the detail @FirePower99 

That endpoint no longer exists, so I'm wondering if the upgrade has been successful? 

Would you be able to please share with me a screenshot of the Version page please? 

Example:
![Screenshot 2024-11-06 at 13 16 18](https://github.com/user-attachments/assets/91409e49-ae43-4aeb-8e3d-692dce1699ec)

**Reply by melohagan** [reply]
It could also be worth clearing any browser cache in case the UI has not updated properly.

**Comment by FirePower99** [comment]
Thanks @melohagan 

<img width="734" alt="image" src="https://github.com/user-attachments/assets/11c976a0-71d9-400d-b8b1-00a6ae6476b4">

**Comment by samwho** [comment]
Hey @FirePower99, re: the error you saw about having to update your version of CouchDB. We found what we believe to be a race condition with how we start up the `budibase/budibase` Docker image. We think we've fixed it in 3.1.2, so if you stay on 3.1.0 you may still see that error sometimes if your container restarts. I'd recommend going up to 3.1.2 to stop seeing that.

We're still looking into the user group data error, thanks for your patience 🙏

**Comment by FirePower99** [comment, marked as answer]
Thanks @melohagan 

I think I have now managed to upgrade to 3.1.2 - This doesn't seem to be an issue any longer.

I also reviewed my docker-compse.yaml and found one of dev's had a custom folder mounted to volume:  - /home/budibase/builder:/app/builder

I have removed this custom volume and now can confirm I can now see the users in the my user directory, and I no longer receive the error.

I apologise for this and thank you so much for your assistance in helping me to troubleshoot.


Have a great rest of your day.

**Reply by melohagan** [reply]
Great to hear, thanks for letting us know.
